inventory/multi_ec2.py

```diff
@@ -59,20 +59,18 @@ def __init__(self):
         if self.args.cache_only:
             # get data from disk
             result = self.get_inventory_from_cache()
+            if result and self.is_cache_valid():
+                return
```

Comment:
What happens if we ask for cache-only and the cache isn't valid?

That doesn't seem to be handled here.